app/controllers/staff/transactions_controller.rb

```diff
@@ -81,6 +82,17 @@ def id
     params[:id]
   end
 
+  def activity
+    @activity ||= Activity.find(activity_id)
+  end
+
+  def associate_transaction_to_submission
+    organisation = activity.organisation
+    fund = activity.fund? ? activity : activity.parent_activities.select { |a| a.level = :fund }
```

Comment:
I'm wondering if there's a better way of finding the active submission for the fund + organisation 🤔

Reply:
I can't immediately think of a single-query way of doing this, because there's an unknown number of jumps from the current activity to its containing fund. If the number of jumps was known, you could do this with joins, but as it is you're essentially following a linked list until you find the thing you're looking for. Loading all of `parent_activities` isn't wasteful because the fund will be the last one in the chain, right?